presto-spark-base/src/main/java/com/facebook/presto/spark/execution/NativeExecutionProcess.java

```diff
@@ -0,0 +1,198 @@
+        this.nodeConfig = requireNonNull(nodeConfig, "nodeConfig is null");
+        this.connectorConfig = requireNonNull(connectorConfig, "connectorConfig is null");
+        this.nativeExecutionTask = new NativeExecutionTask(session, planFragment, sources, this.workerClient, tableWriteInfo, executor, updateScheduledExecutor, taskManagerConfig);
+    }
+
+    public void close()
+    {
+        if (process.isAlive()) {
+            process.destroy();
+        }
+    }
+
+    private URI getBaseUriWithPort(URI baseUri, int port)
+    {
+        return uriBuilderFrom(baseUri)
+                .port(port)
+                .build();
+    }
+
+    private int getAvailableTcpPort()
+            throws IOException
+    {
+        int port = NATIVE_PROCESS_DEFAULT_PORT;
+        ServerSocket socket = new ServerSocket(0);
+        port = socket.getLocalPort();
+        socket.close();
+        return port;
+    }
+
+    private String getNativeExecutionCatalogName(Session session)
+    {
+        checkArgument(session.getCatalog().isPresent(), "Catalog isn't set in the session.");
+        return session.getCatalog().get();
+    }
+
+    private void populateConfigurationFiles(String configBasePath)
+            throws IOException
+    {
+        systemConfig.setHttpServerPort(port);
+        WorkerProperty.populateProperty(systemConfig.getAllProperties(), Paths.get(configBasePath, WORKER_CONFIG_FILE));
+        WorkerProperty.populateProperty(nodeConfig.getAllProperties(), Paths.get(configBasePath, WORKER_NODE_CONFIG_FILE));
+        WorkerProperty.populateProperty(
+                connectorConfig.getAllProperties(),
+                Paths.get(configBasePath, format("%s%s.properties", WORKER_CONNECTOR_CONFIG_FILE, getNativeExecutionCatalogName(session))));
+    }
+
+    public String getProcessWorkingPath(String path)
+    {
+        File absolutePath = new File(path);
+        File workingDir = new File(SparkFiles.getRootDirectory());
+        if (!absolutePath.isAbsolute()) {
+            absolutePath = new File(workingDir, path);
+        }
+
+        if (!absolutePath.exists()) {
+            throw new RuntimeException(format("File doesn't exist %s", absolutePath.getAbsolutePath()));
+        }
+
+        return absolutePath.getAbsolutePath();
+    }
+
+    /**
+     * Starts the external native execution process. The method will be blocked by connecting to the native process's /v1/info endpoint with backoff retries until timeout.
+     */
+    public void start()
+            throws ExecutionException, InterruptedException, IOException
+    {
+        String executablePath = getProcessWorkingPath(SystemSessionProperties.getNativeExecutionExecutablePath(session));
+        String configPath = Paths.get(executablePath, String.valueOf(port)).toAbsolutePath().toString();
+
+        populateConfigurationFiles(configPath);
+        ProcessBuilder processBuilder = new ProcessBuilder(executablePath, "--v", "1", "--etc_dir", configPath);
```

Comment:
Would be nice if we can have the commands configurable through. That way we can enable/disable any additional features dynamically.